src/perf/lib/HpsClient.h

```diff
@@ -33,7 +33,7 @@ struct HpsWorkerContext {
         QuicZeroMemory(&LocalAddrs, sizeof(LocalAddrs));
         QuicEventInitialize(&WakeEvent, FALSE, TRUE);
     }
-    ~HpsWorkerContext() {
+    void WaitForWorker() {
         if (ThreadStarted) {
```

Comment:
Don't do it this way. Instead make a new WaitForWorker function that just sets the event and waits on the thread. Keep the destructor as is.